app/models/queue_config.rb

```
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+class QueueConfig
+  include ActiveModel::Model
+
+  attr_accessor :organization
+
+  def initialize(args)
+    super
+
+    if !organization&.is_a?(Organization)
+      fail(
+        Caseflow::Error::MissingRequiredProperty,
+        message: "organization property must be an instance of Organization"
+      )
+    end
```

I want `QueueConfig.new` to fail immediately if there is no organization argument (or it is not actually an `Organization`) and I could not wrangle `ActiveModel` validations to do that for me. Would love to move this back to a validation if it is possible.

---

We can check the validity in a custom validator, but I'm not sure how to have the validator raise the exception immediately upon validation instead of doing the whole `obj.new; fail(SomeError) unless obj.valid?` dance.

I'd like to avoid the pattern of checking validity in the calling code, in favor of capturing that logic within the code that is called itself. I prefer to throw the exception as early as we know we we've encountered an error, preserve the original error as long as possible, and handle the exception when we are ready to (which in this case I don't think would be until we hit [the existing logic in `ApplicationController`](https://github.com/department-of-veterans-affairs/caseflow/blob/45725534fe7a631dab22143d9544668031ed946b/app/controllers/application_controller.rb#L12) to return the error to the front-end).

What I'd like to avoid:
```ruby
  # app/models/generic_queue.rb
  def config
    if user.is_a?(Organization)
      queue_config = QueueConfig.new(organization: user)
      if !queue_config.valid?
        fail(SomeError)
      end
      queue_config.to_h
    else
      {}
    end
  end
```

---

I think the `after_initialize` callback might be what you're after? 

https://guides.rubyonrails.org/v2.3.11/activerecord_validations_callbacks.html#after-initialize-and-after-find

---

I think that is only available for `ActiveRecord` classes and not `ActiveModel` classes.